### PR TITLE
Make DBT dependency optional

### PR DESCRIPTION
**Pull request by erezsh**
Fixes Issue #412

### CODE REVIEW

**Comment by erezsh** [comment]
@dlawin Any idea why these tests are failing?

**Comment by dlawin** [comment]
> @dlawin Any idea why these tests are failing?

Hopefully it's clear from this: [21804e5](https://github.com/datafold/data-diff/pull/421/commits/21804e5d9553e2d1af32e28ba71b3903102b6411)

Need to use the mock_self as the mock, since that's how the methods are being called now. No need to mock the package

**Comment by dlawin** [comment]
> > @dlawin Any idea why these tests are failing?
> 
> Hopefully it's clear from this: [21804e5](https://github.com/datafold/data-diff/pull/421/commits/21804e5d9553e2d1af32e28ba71b3903102b6411)
> 
> Need to use the mock_self as the mock, since that's how the methods are being called now. No need to mock the package (and a package level mock won't override the self.blah methods now)

I can fix the remaining ones later today

File: data_diff/dbt.py [hunk 1]
```diff
@@ -3,14 +3,21 @@
...
-from dbt_artifacts_parser.parser import parse_run_results, parse_manifest
-from dbt.config.renderer import ProfileRenderer
+
+def import_dbt():
+    try:
+        from dbt_artifacts_parser.parser import parse_run_results, parse_manifest
+        from dbt.config.renderer import ProfileRenderer
+        import yaml
+    except ImportError:
+        raise RuntimeError("Could not import 'dbt' package. You can install it using 'pip install data-diff[dbt]'.")
```
Comment (williebsweet) [inline data_diff/dbt.py:18]: Does this need to be `pip install 'data-diff[dbt]'`?

**Comment by erezsh** [comment]
@dlawin Nice, thank you!

File: data_diff/dbt.py (hunk 1 above)
Comment (williebsweet) [reply data_diff/dbt.py:18]: I disagree - we should address this particular case in this particular issue. No reason to create an error that is not useful.

We can address the other instances in follow up issues but we should address this one in this issue.

Comment (erezsh) [reply data_diff/dbt.py:18]: So what do you suggest? There is no single form of writing it that will work for everyone.

Comment (williebsweet) [reply data_diff/dbt.py:18]: Does `pip install 'data-diff[dbt]'` work for some people and `pip install 'data-diff[dbt]'` work for others?

Comment (williebsweet) [reply data_diff/dbt.py:18]: I don't understand the pushback. I know we have an issue though. If I run `pip install data-diff[dbt]` - I get the following "no matches found: data-diff[dbt]". Which may leave users confused and without a next step.

Comment (erezsh) [reply data_diff/dbt.py]: Done.

**Review by williebsweet — APPROVED** [review]

**Pull request merged**